### DISCUSSION TITLE
How to create Reader for Keystatic Cloud content?

### DISCUSSION THREAD

**Discussion by NeeshSamsi** [Thinkmill/keystatic#1240] [Q&A]
Hello! Used keystatic with github for a while now. Just started a new small project for a client and since we're not storing any images, I thought Keystatic Cloud would be the way to go.

Onboarding was great, got everything set including my schemas and added some data into the UI.

I am unable to retrieve the data tho since the [Reader API docs](https://keystatic.com/docs/reader-api) only have guides for Local and Github Storage. I tried using the local method but I am receiving `null`. And to try the Github Reader, I don't know what repo I would add.

Such an unnecessary roadblock, it's brought me to a complete halt. Would appreciate any help.

**Comment by amirzezo201** [comment]
Yesterday same thing happened to me as well i was using keystatic reader api in astro server action and it was returning null the only way was to use the Astro getcollection or getentry that works fine
